tests/RobotFramework/tests/cumulocity/software.robot

```diff
@@ -16,6 +16,22 @@ Install software via Cumulocity
     Operation Should Be SUCCESSFUL           ${OPERATION}    timeout=60
     Device Should Have Installed Software    c8y-remote-access-plugin
 
+# TODO: make it not flaky
+tedge-agent should terminate on SIGINT while downloading file
+    # we download a file which is 500M, but tmpfs at /tmp is only 64M, so we
+    # have to change tmp.path to be able to store the download
+    Execute Command                          chmod 777 /root
+    Execute Command                          tedge config set tmp.path /root
+    Restart Service                          tedge-agent
+    ${OPERATION}=    Install Software        test-very-large-software,1.0,https://t493319102.eu-latest.cumulocity.com/inventory/binaries/28057693
+
+    # waiting for the download to start (so, for "Downloading: ...") to appear
+    # in the log, but I have no clue how to do "wait until log contains ..."
```

Comment:
One possibility is to use the Keyword "Wait Untill Keyword Succeeds", here one example:
``` 
Wait Until Log Contains
    [Arguments]  ${expected_message}  ${timeout}
    Wait Until Keyword Succeeds  ${timeout}  2s
        Run Keyword And Return Status  Should Log Contain  ${expected_message} 

Should Log Contain
    [Arguments]  ${expected_message}
    ${logs}  Get Logs
    Log Many  ${logs}  level=INFO  console=yes
    Log Many  ${logs}  level=DEBUG  console=yes
    Should Contain  ${logs}  ${expected_message}
```
ping me we can try it together if you want

Reply:
I think it will be better to write a keyword in python to facilitate this, as I want to avoid doing too much programming in RobotFramework language if we can help it...And I don't think this will be the last time we have this use-case.

So in summary, I can write a new assertion (which will be exposed as a RobotFramework keyword).

Reply:
I've pushed some updates to the test to add a new keyword to check the presence of log entries (and dynamically wait for them). Just waiting for the CI to run to confirm, however I ran the test 50 times locally and no signs of flakiness.